### ISSUE TITLE
hN accessibility concerns

### ISSUE THREAD

**Issue by Alex-Jordan** [PreTeXtBook/pretext#890]
* Each HTML section needs an hN for accessibility. At a minimum, untitled introductions (and conclusions?) have no hN. One possibility is to use hN with a class="assistive" (like the skip to main content link)
* Validator suggests each HTML article needs an hN. Investigate this.
* a PTX section is h2, even when chunking at the section level. So you can have a page for a section where what looks like it should be the h1 is actually an h2. I believe it is a problem to have no h1 at all in this situation.
* if a book is in parts, the TOC might have both the parts and the chapters as h2. Instead, in the TOC, parts should be h2, chapters h3, sections h4. (If the book has no parts, research if we then need chapter to be h2, section h3, subsection h4, or if you may have chapter h3, section h4, subsection h5.)
* Lots of things are h6, probably by necessity since a chapter page could use h1 and go all the way down to a paragraphs at h5, and within that you could have figures, etc. Is it an accessibility concern to have h6 header for something within something that has an h6 header? For example, a figure h6 within a case h6 within a proof h6 within a theorem h6?

**Comment by rbeezer** [comment]
Thanks for the summary, Alex.

`paragraphs/theorem/proof/case` is a good example!  (Do we put headings on `figure`?  It will have a `figcaption`.)  If it was only buried in a `chapter/section` we'd have just enough room (six levels).

If we want to have **every** HTML page go strictly `h1` to `h-max` I think I can make that happen without too much surgery.  At `h6` I think we just need to punt and top out there (i.e. `h6` on all subsequently deeper headings).

The ToC was just hacked to accomodate parts.  More care should be part of the current re-factoring.  (In other words, too hard to fix now, and a waste of time.)

I'm fine with suggestions for `introduction`.  Alex has suggested that `objectives` might be more "aside-like".  Here's an example of some marginal material I just found recently.  Try the `+` inside discs.
http://udleditions.cast.org/INTRO,all_about_coyotes.html

**Comment by rbeezer** [comment]
The warnings about headings from the Nu HTML Checker link to the following, 
which is a succinct summary:

https://www.w3.org/wiki/HTML/Usage/Headings/Missing

**Comment by rbeezer** [comment]
We were already onto this one at #729

**Comment by davidfarmer** [comment]
Probably covered by recent work of Alex J. If not, make a new issue.
